plugins/modules/lxd_container.py

```diff
@@ -469,10 +469,14 @@ def __init__(self, module):
 
         # LXD Rest API provides additional endpoints for creating containers and virtual-machines.
         self.api_endpoint = None
-        if self.type == 'container':
-            self.api_endpoint = '/1.0/containers'
-        elif self.type == 'virtual-machine':
-            self.api_endpoint = '/1.0/virtual-machines'
+        try:
+            self.api_endpoint = '/1.0/instances'
+            self.client.do('GET', self.api_endpoint)
```

Comment:
This function can raise an exception for quite a few different reasons. Why not pass `ok_error_codes=[404]` and check the return value whether it's an error?

Also a different question: how expensive is a GET to `/1.0/instances`? Maybe instead do a `HEAD` or `OPTIONS` HTTP request.